Swiftfin/Views/UserProfileImagePicker/UserProfileImage.swift

```diff
@@ -0,0 +1,102 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2024 Jellyfin & Jellyfin Contributors
+//
+
+import Defaults
+import JellyfinAPI
+import SwiftUI
+
+struct UserProfileImage: View {
```

Comment:
If we're going to make some kind of central view to handle showing a user profile image we should make it applicable to the following areas: `UserProfileRow`, `UserRow`, `PublicUserRow`. (Although I think some reflection on the names of those views is warranted).

All of these show a user profile image but some are currently incomplete due to design (like the background colors and placeholder/failure views) having been done after their implementations. We should probably make a component view for showing a user profile image in a circle. That would change _this_ specific view to something like `UserProfileHeroImage`, since this is a hero view of editing or presenting a user.

I think the caching issues come down to which pipeline is used. The `branding` pipeline is a separate "cache" used for images that should be "offline", like local user profile images and the server splash screen (at least by choice). Not all user images, especially when managing them as a server administrator and they haven't signed in locally, should be considered "something to be seen offline" and use the default pipeline, which is like a normal memory cache.

You can see the branding pipeline results in `Files/On My iPhone/Swiftfin iOS/Cache/org.jellyfin.swiftfin.branding`.

Reply:
I can work on this! 

On a similar note, is there a way to use: `RedrawOnNotificationView(.didChangeUserProfile)` where I can take the `.didChangeUserProfile` payload into account? I pass the `userID` as the payload and it would be nice to reserve redrawing for matching `userIDs`

Reply:
I've added the capability for `RedrawOnNotificationView` to filter based on the payload if necessary.

Reply:
I centralized all user icons to `UserProfileHeroImage`. I made a `UserEditableHeroImage` for this `pencil` + `UserProfileHeroImage` where the `confirmationDialog` pops up to replace/delete:

<img width="180" alt="Screenshot 2024-12-19 at 10 20 17" src="https://github.com/user-attachments/assets/7d1b08f2-7386-4046-ac64-424943630a7d" />

I updated the verbiage on this to better reflect caching:

![simulator_screenshot_C9E6F7F8-F0FE-4B96-AACA-FED83995629C](https://github.com/user-attachments/assets/ee679af8-c7a5-4b25-be3a-ce6cd0aa6a90)

I hope this was what you meant! Let me know if there are changes/adjustments that are needed for this. This is also used for tvOS so merging with some other branches might start getting a bit weird.